pkg/execution/driver/httpdriver/httpdriver_test.go

```diff
@@ -34,7 +34,7 @@ func TestRedirect(t *testing.T) {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch count {
 		case 8:
-			require.Equal(t, http.MethodPost, r.Method)
+			require.Equal(t, http.MethodGet, r.Method)
```

Comment:
This is scary.

A user might configure a gateway to use a 303.  This converts POSTs to GETs.  It's what this test covers.

We don't want that.  We actually ignore the HTTP spec here.  We do that because _we know what the intent of our request is_, and it *must* use the requested HTTP method after any redirects, instead of what the redirect asked for.

Reply:
That isn't true. If a redirect tells us to use a `GET` then we _must_ use a `GET` after the redirect. Modal switches to a `GET` on the redirect:
1. We send `POST` to execute Inngest function hosted in Modal.
2. After 150 seconds, Modal sends a `303` redirect (which is always a `GET` by the HTTP spec).
3. If we send a `POST` after the redirect then Modal returns a `400` due an unsupported method. It only works if we send a `GET`.